compiler-rt/lib/sanitizer_common/sanitizer_platform_interceptors.h

```diff
@@ -492,7 +492,8 @@
 #define SANITIZER_INTERCEPT_PVALLOC (SI_GLIBC || SI_ANDROID)
 #define SANITIZER_INTERCEPT_CFREE (SI_GLIBC && !SANITIZER_RISCV64)
 #define SANITIZER_INTERCEPT_REALLOCARRAY SI_POSIX
-#define SANITIZER_INTERCEPT_ALIGNED_ALLOC (!SI_MAC)
+#define SANITIZER_INTERCEPT_ALIGNED_ALLOC (!(SI_MAC) || SI_MAC_DEPLOYMENT_AT_LEAST_10_15)
+#define SANITIZER_INTERCEPT_SHARED_MUTEX (!(SI_MAC) || SI_MAC_DEPLOYMENT_AT_LEAST_10_12)
```

Comment:
I agree that we should add `SANITIZER_INTERCEPT_ALIGNED_ALLOC` here, because it feels like it is re-usable in other sanitizers and applies fairly generally. But I don't feel the same way about `SANITIZER_INTERCEPT_SHARED_MUTEX`, for a couple of reasons:

1. there's technically actually no interception of `shared_mutex` methods anywhere, and
2. it's only really a concern for the radsan tests.

My vote would be to move this definition to somewhere in `radsan_test.cpp`, and rename it to describe what it guards more accurately - perhaps something like `RADSAN_TEST_SHARED_MUTEX`?